### DISCUSSION TITLE
Can I import selectors` { get, set } functions directly?

### DISCUSSION THREAD

**Discussion by schmidlidev** [facebookexperimental/Recoil#1150] [Q&A]
Selectors expose `get` and `set` by passing them as parameters to your selectors` functions. Are these functions exposed by the Recoil package in any other way?

I have somewhat complex interdependent state. I have some writeable selectors that read from and write to many other atoms. I would like to extract out some of this logic for re-use. For example, from within several of my selector's `set:` functions, I want to call the same `doAction()` function.

Trivially, I can have my re-useable function accept {get, set} in its parameters, and just forward the references from within the selector:

```ts
import { GetRecoilValue, SetRecoilState } from 'recoil';

interface Opts {
  get: GetRecoilValue,
  set: SetRecoilState
}

// Extracted and re-usable function:
function doAction({get, set}: Opts, someParam: number) {
  // Do some action using get/set on other atoms.
}

// Example use:
const mySelector = selector({
  key: 'mySelector',
  set: ({get, set}, newValue) => {
    // Some logic
    const something = doAction({get, set}, 12345) // Using the action 
    // Some more logic
  }
})
```

But it would just be cleaner if my extracted function could simply import `get` and `set` from Recoil, so that they don't need to be passed in as parameters. Is this possible?

**Comment by rbaxter08** [comment]
I don't think so, because the `get` and `set` wouldn't have access to the store provided by the RecoilRoot.  In a way this is basically asking if Recoil can work outside of React, which I don't believe it can.

For what it's worth, your solution seems fine to me.  Maybe it would make more sense if `doAction` was a selector instead of just a function, but would have to see more code.

**Reply by drarmstr** [reply]
If `doAction()` is just extracted to normalize redundant code getting/setting atoms, that's fine.  But keep in mind that selectors shouldn't do side-effects from their get/set methods.
